src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/PackageService.java

```diff
@@ -2,98 +2,98 @@
 
 
 import com.example.ludogoriesoft.lukeriaerpapi.dtos.PackageDTO;
-import com.example.ludogoriesoft.lukeriaerpapi.exeptions.ApiRequestException;
-import com.example.ludogoriesoft.lukeriaerpapi.mappers.PackageMapper;
 import com.example.ludogoriesoft.lukeriaerpapi.models.Package;
+import com.example.ludogoriesoft.lukeriaerpapi.repository.CartonRepository;
 import com.example.ludogoriesoft.lukeriaerpapi.repository.PackageRepository;
 import io.micrometer.common.util.StringUtils;
+import jakarta.validation.ValidationException;
 import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.data.crossstore.ChangeSetPersister;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
-import java.util.Optional;
 
 @Service
 @AllArgsConstructor
 public class PackageService {
     private final PackageRepository packageRepository;
-    private final PackageMapper mapper;
-
-    public PackageDTO toDTO(Package packageEntity) {
-        return mapper.toDto(packageEntity);
-    }
-
-    public Package toEntity(PackageDTO packageDTO) {
-        return mapper.toEntity(packageDTO);
-    }
-
+    private final CartonRepository cartonRepository;
+    private final ModelMapper modelMapper;
 
     public List<PackageDTO> getAllPackages() {
-        List<Package> packages = packageRepository.findAll();
-        return packages
-                .stream()
-                .map(this::toDTO)
-                .toList();
+        List<Package> laptops = packageRepository.findByDeletedFalse();
+        return laptops.stream().map(package1 -> modelMapper.map(package1, PackageDTO.class)).toList();
     }
 
-    public PackageDTO getPackageById(Long id) {
-        Optional<Package> optionalPackage = packageRepository.findById(id);
-        if (optionalPackage.isEmpty()) {
-            throw new ApiRequestException("Package with id: " + id + " Not Found");
-        }
-        return toDTO(optionalPackage.get());
+    public PackageDTO getPackageById(Long id) throws ChangeSetPersister.NotFoundException {
+        Package package1 = packageRepository.findByIdAndDeletedFalse(id).orElseThrow(ChangeSetPersister.NotFoundException::new);
+        return modelMapper.map(package1, PackageDTO.class);
     }
 
     public PackageDTO createPackage(PackageDTO packageDTO) {
```

Comment:
We already have those validations in the Package model. Let's do what Emo mentioned (https://github.com/Ludogorie-Soft/lukeria-erp-api/pull/9#discussion_r1260304069):
_The validations should be moved from the model to the DTO_

Reply:
I meant just to move the Package model validations to the PackageDTO. I see that you did this for all the Models and DTOs. Usually, this should be done in a separate PR. For example, Berkant currently is working on Product. He could do that. However, let's do it here. We just need to make sure that everything is OK when merging and tell to Berkant that we did this for him.